### PR TITLE
Add various weighted heuristics to imrove bot moves

### PR DESCRIPTION
**Pull request by jonathan-marmelab**
This closes the following user story: https://trello.com/c/FyVjSW4q/57-as-robert-i-want-to-get-better-advice

### CODE REVIEW

File: domain/internal/model/place_tile_hint.go [hunk 1]
```diff
@@ -69,3 +74,41 @@ func (b *Board) GetPlaceTileHint() (*Board, *PlaceTileHint) {
 
 	return bestBoardCopy, bestPlaceTileHint
 }
+
+// getPlaceTileScore returns the board score that optimizes the number of available tiles for the player while minimizing the number of available tiles for the next player
+func (b *Board) getPlaceTileScore() float64 {
+	var (
+		accessibleTiles, _                      = b.GetAccessibleTiles()
+		accessibleTilesCount                    = float64(len(accessibleTiles))
```
Comment (guilbill) [inline domain/internal/model/place_tile_hint.go:82]: Float?

File: domain/ai/main.go [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+outer:
+	for round := 0; round < 100; round++ {
+		fmt.Printf("----------\n")
+		fmt.Printf("[Round #%d] Start\n", round)
+
+		board, _ := model.NewBoard(7, 4)
+
+		// We want to keep some weights for optimization
+		for _, player := range board.Players {
+			switch player.Color {
```
Comment (Nitix) [inline domain/ai/main.go:21]: I don't really understand the principle of switching over colors. You want the IA to have multiple algorithm? 
Also is colors always the same for the IA?

**Review by Nitix — APPROVED** [review]
LGTM

File: domain/ai/main.go (hunk 2 above)
Comment (jonathan-marmelab) [reply domain/ai/main.go:21]: This file is used for AI weight training, and this switch allows to keep some weights for the next round to optimize the training

**Pull request merged**